scripts/get-git-tag-name.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+# This script derives a git tag name from the version fields found in a given Go
+# file. It also checks if the derived git tag name is a valid SemVer compliant
+# version string.
+
+# get_git_tag_name reads the version fields from the given file and then
+# constructs and returns a git tag name.
+get_git_tag_name() {
+  local file_path="$1"
+
+  # Check if the file exists
+  if [ ! -f "$file_path" ]; then
+      echo "Error: File not found at $file_path" >&2
+      exit 1
+  fi
+
+  # Read and parse the version fields.
+  local app_major
+  app_major=$(grep -oP 'AppMajor\s*uint\s*=\s*\K\d+' "$file_path")
+
+  local app_minor
+  app_minor=$(grep -oP 'AppMinor\s*uint\s*=\s*\K\d+' "$file_path")
+
+  local app_patch
+  app_patch=$(grep -oP 'AppPatch\s*uint\s*=\s*\K\d+' "$file_path")
+
+  local app_status
+  app_status=$(grep -oP 'AppStatus\s*=\s*"([a-z]*)"' "$file_path" | cut -d '"' -f 2)
+
+  local app_pre_release
+  app_pre_release=$(grep -oP 'AppPreRelease\s*=\s*"([a-z0-9]*)"' "$file_path" | cut -d '"' -f 2)
+
+  # Parse the GitTagIncludeStatus field.
+  gitTagIncludeStatus=false
+
+  if grep -q 'GitTagIncludeStatus = true' "$file_path"; then
+      gitTagIncludeStatus=true
+  elif grep -q 'GitTagIncludeStatus = false' "$file_path"; then
+      gitTagIncludeStatus=false
+  else
+      echo "Error: GitTagIncludeStatus is not present in the file."
+      exit 1
+  fi
+
+  # Construct the git tag name with conditional inclusion of app_status and
+  # app_pre_release
+  tag_name="v${app_major}.${app_minor}.${app_patch}"
+
+  # Append app_status if gitTagIncludeStatus is true and app_status is not empty
+  if [ "$gitTagIncludeStatus" = true ] && [ -n "$app_status" ]; then
+      tag_name+="-${app_status}"
+  fi
+
+  # Append app_pre_release if it is not empty
+  if [ -n "$app_pre_release" ]; then
+      tag_name+="-${app_pre_release}"
+  fi
+
+  echo "$tag_name"
+}
+
+# check_git_tag_name checks if the given git tag name is a valid SemVer
+# compliant version string.
+check_git_tag_name() {
```

Comment:
Do we even need this function anymore? Since we assemble the git tag in this script, we know it should produce a valid semver tag (and if either of the string variables contain something breaking, it would panic when starting the binary, so unit tests would break immediately).

Reply:
Yes I think you're right.